kong/conf_loader.lua

```diff
@@ -235,6 +236,10 @@ local CONF_INFERENCES = {
 
   lua_ssl_verify_depth = { typ = "number" },
   lua_socket_pool_size = { typ = "number" },
+  role = { enum = { "proxy", "admin", "traditional", }, },
```

Comment:
I don't want to bike-shed on this but are `admin` and `proxy` accurate?
They are already overloaded terms in Kong's terminology. `proxy` is okay but I'm not sure if that's what we want Kong to limit to, especially when we take the enterprise product into consideration as well.

I'm sure there was more discussion on this before that I might not be aware of.

Reply:
Yes I believe this was used on the product design side. @thibaultcha thoughts?

Reply:
'data_plane' and 'control_plane' or simply 'dp' and 'cp' sounds more straightforward.